### DISCUSSION TITLE
Generating Open API spec using isc.rest

### DISCUSSION THREAD

**Discussion by dangoldmanvt1** [intersystems/isc-rest#13] [Q&A] [answered]
I am starting to migrate from using apps_rest to  use isc.rest for a project and we need to send an open api spec to the front end developers for our api.  I am trying to get an example working with the sample phonebook app.  Does anyone have an example of generating an open api spec using the isc.rest framework ? Thanks Dan

**Comment by isc-tleavitt** [comment, marked as answer]
We still need to flesh out the docs on OpenAPI spec generation in itself, but the best way to work with it is through https://github.com/intersystems/isc-ipm-js . This also has a sample of how to generate OpenAPI at [https://github.com/intersystems/isc-ipm-js/blob/main/cls/pkg/isc/ipm/js/...](https://github.com/intersystems/isc-ipm-js/blob/main/cls/pkg/isc/ipm/js/openApiProcessor.cls#L41) and following.
